### ISSUE TITLE
Instaloader Aborts with 302 Redirect to Login Page

### ISSUE THREAD

**Issue by milliorn** [instaloader/instaloader#2320]
I am encountering an issue when using Instaloader to download profile data. The command used is as follows:

`instaloader profile milliorn --filename-pattern="{filename}" --latest-stamps --no-videos --no-video-thumbnails --no-captions --no-metadata-json --no-compress-json --no-profile-pic --max-connection-attempts=1 --abort-on=302,400,401,429 --request-timeout=1`

The output indicates that the download was aborted due to a "302 Found" which I have never encountered using this app. I just updated my copy of this repo today.

```
Using latest stamps from /home/milliorn/.config/instaloader/latest-stamps.ini.

Download aborted: Query to https://www.instagram.com/web/search/topsearch/ responded with "302 Found" redirect to https://www.instagram.com/accounts/login/?next=/web/search/topsearch/%3Fcontext%3Dblended%26query%3Dprofile%26include_reel%3DFalse%26__a%3D1.
```
[https://www.instagram.com/web/search/topsearch/?context=blended&query=profile&include_reel=False&__a=1](url)

Could you please assist in resolving this issue? Thank you for your help.

**Comment by ekalin** [comment]
Well, you did ask for it to abort on a 302 response code, so it did.

302 is not an error, just an indication that the query should be repeated at another address. instaloader will follow the redirect, unless told to abort (which you did).

In this case it wouldn't really have helped, IG has logged you out and used a 302 status to get the user to the login screen.

This is pretty much unavoidable - eventually IG detects some behavior that it considers unusual and logs you out, maybe requesting you to solve a captcha. Just log in again and continue.

It's better to not use your real account, sometimes IG blocks accounts.

**Comment by milliorn** [comment]
> Well, you did ask for it to abort on a 302 response code, so it did.
> 
> 302 is not an error, just an indication that the query should be repeated at another address. instaloader will follow the redirect, unless told to abort (which you did).
> 
> In this case it wouldn't really have helped, IG has logged you out and used a 302 status to get the user to the login screen.
> 
> This is pretty much unavoidable - eventually IG detects some behavior that it considers unusual and logs you out, maybe requesting you to solve a captcha. Just log in again and continue.
> 
> It's better to not use your real account, sometimes IG blocks accounts.

This is the some of the reason why I came here. It did force me to log out (in the browser). I did log back in. I check my account status and nothing is restricted, all green.  So I am confused what is going on in the CLI, since it differs from what I see in the browser?

**Comment by ekalin** [comment]
That's the way it is, unfortunately... IG somehow detects "unusual activity" or "scraping behavior" and occasionally logs you out. Sometimes one just needs to click on a "Dismiss" button, sometimes a captcha is presented, sometimes a full login is required.

And before you ask, no one knows exactly what triggers this detection. And no, it's not (just) the speed of requests, despite what some other issues here might suggest. Maybe it plays a role, but simply slowing down the downloads is not enough to keep IG happy.

**Comment by milliorn** [comment]
> That's the way it is, unfortunately... IG somehow detects "unusual activity" or "scraping behavior" and occasionally logs you out. Sometimes one just needs to click on a "Dismiss" button, sometimes a captcha is presented, sometimes a full login is required.
> 
> And before you ask, no one knows exactly what triggers this detection. And no, it's not (just) the speed of requests, despite what some other issues here might suggest. Maybe it plays a role, but simply slowing down the downloads is not enough to keep IG happy.

tl;dr the account isn't banned or restricted, with typical normal usage. However, IG will has now sniffed this account out and will block it from CLI activity. Is that what I am reading?

**Comment by ekalin** [comment]
Maybe it will, maybe it won't. But as I said before, do not use your real account with instaloader - there is the risk you'll be blocked. Use a throwaway account, if you loose it, no big deal.
